From: Miklos Vajna <vmiklos@frugalware.org>
Date: Fri, 3 Nov 2006 16:33:38 +0000 (-0500)
Subject: [PATCH] fix for udev-094
X-Git-Url: http://git.kernel.org/?p=linux%2Fpcmcia%2Fpcmciautils.git;a=commitdiff_plain;h=3a5217492c1f95711b30f4e27c51563dff6ceb3e

[PATCH] fix for udev-094

from udev's release notes:

"udev 094
========
The built-in MODALIAS key and substitution is removed.
(...)
udev 080
========
(...)
MODALIAS and $modalias is not needed and will be removed from one of
the next udev versions, replace it in all rules with ENV{MODALIAS} or
the sysfs "modalias" value."

here is a patch to fix this issue

Signed-off-by: Miklos Vajna <vmiklos@frugalware.org>
Signed-off-by: Dominik Brodowski <linux@dominikbrodowski.net>
---

diff --git a/udev/rules-base b/udev/rules-base
index c6d14de..6ed3a27 100644
--- a/udev/rules-base
+++ b/udev/rules-base
@@ -3,12 +3,12 @@
 # are so broken that we need to read out random bytes of it
 # instead of the manufactor, card or product ID. Then the
 # matching is done in userspace.
-ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="?*", \
+ACTION=="add", SUBSYSTEM=="pcmcia", ENV{MODALIAS}=="?*", \
 		RUN+="/sbin/pcmcia-check-broken-cis"
 
 # However, the "weak" matching by func_id is only allowed _after_ modprobe
 # returns, so that "strong" matches have a higher priority.
-ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="?*", \
+ACTION=="add", SUBSYSTEM=="pcmcia", ENV{MODALIAS}=="?*", \
 		RUN+="/bin/sh -c 'echo 1 > /sys/$devpath/allow_func_id_match'"
 
 # PCMCIA sockets:
diff --git a/udev/rules-modprobe b/udev/rules-modprobe
index ea1f44a..a13b2ad 100644
--- a/udev/rules-modprobe
+++ b/udev/rules-modprobe
@@ -1,3 +1,3 @@
-# modprobe $modalias loads all possibly appropriate modules
-ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="?*", \
-		RUN+="/sbin/modprobe $modalias"
+# modprobe $env{MODALIAS} loads all possibly appropriate modules
+ACTION=="add", SUBSYSTEM=="pcmcia", ENV{MODALIAS}=="?*", \
+		RUN+="/sbin/modprobe $env{MODALIAS}"
